### PR TITLE
BaseButton: make input related information in the tooltip optional

### PR DESCRIPTION
**Pull request by lupoDharkael**
solves https://github.com/godotengine/godot/issues/23617

### CODE REVIEW

**Comment by akien-mga** [comment]
Won't that remove all shortcut hints from the editor buttons?

**Comment by lupoDharkael** [comment]
I didn't think about it. I don't think a tooltip should be imposed but the lack of automatic editor tooltips about the shortcuts is inconvenient.
What about having conditional show of the shortcuts in the tooltip?

**Comment by CowThing** [comment]
Yeah that's true, I forgot about that, the editor does use this for shortcut hints in buttons. I think the real problem is that InputEventAction's get_text() function returns a string that isn't any help for shortcut hints, because it doesn't actually show which buttons trigger the action, and has a lot of extra text on it. InputEventKey's get_text() function just returns the key and modifiers, with no extra text, making it much better for hints.

An option to disable the shortcut tooltip would be nice if there's no easy way to show which inputs trigger a specific InputEventAction.

**Comment by lupoDharkael** [comment]
I've reimplemented this PR, now it exposes a variable in the BaseButton class to disable the input related info in the tooltip.
Tell me if you agree with the name of the methods and the variable exposed to gdscript so I can start adding this to the docs in this PR.

File: scene/gui/base_button.cpp [hunk 1]
```diff
@@ -535,6 +547,7 @@ void BaseButton::_bind_methods() {
 	ADD_SIGNAL(MethodInfo("toggled", PropertyInfo(Variant::BOOL, "button_pressed")));
 	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "disabled"), "set_disabled", "is_disabled");
 	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "toggle_mode"), "set_toggle_mode", "is_toggle_mode");
+	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "show_input_tooltip"), "set_show_input_tooltip", "is_show_input_tooltip_enabled");
```
Comment (akien-mga) [inline scene/gui/base_button.cpp]: Maybe `tooltip_with_shortcut_info`?

Comment (akien-mga) [reply scene/gui/base_button.cpp]: Or `add_shortcut_to_tooltip`? It's hard to find something not too long but still descriptive enough :D

Comment (akien-mga) [reply scene/gui/base_button.cpp]: `shortcut_in_tooltip` maybe.

Comment (lupoDharkael) [reply scene/gui/base_button.cpp]: `shortcut_in_tooltip` seems like a good option.

**Comment by lupoDharkael** [comment]
I've added it to the docs.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**